README.md

```diff
@@ -104,17 +99,75 @@ docker-compose build -- builder batch_submitter
 docker-compose start batch_submitter
 ```
 
+Source code changes can have an impact on more than one container.
+**If you're unsure about which containers to rebuild, just rebuild them all**:
+
+```
+cd ops
+docker-compose down
+docker-compose build
+docker-compose up
+```
+
+Finally, **if you're running into weird problems and nothing seems to be working**, run:
+
+```
+cd optimism
+yarn clean
+yarn build
+cd ops
+docker-compose down -v
+docker-compose build
+docker-compose up
+```
+
+#### Viewing docker container logs
 By default, the `docker-compose up` command will show logs from all services, and that
 can be hard to filter through. In order to view the logs from a specific service, you can run:
 
 ```
 docker-compose logs --follow <service name>
 ```
-### Static analysis
 
-To run `slither` locally in `./packages/contracts` do
+### Running tests
+
+Before running tests: **follow the above instructions to get everything built.**
+
+#### Running unit tests
+
+Run unit tests for all packages in parallel via:
+
+```bash
+yarn test
+```
+
+To run unit tests for a specific package:
+
+```bash
+cd packages/package-to-test
+yarn test
+```
+
+#### Running integration tests
+
+Follow above instructions for building the whole stack.
+Build and run the integration tests:
 
+```bash
+cd integration-tests
+yarn build:integration
```

Comment:
```suggestion
cd integration-tests
yarn install
yarn build:integration
```

This is necessary (just checked to confirm), because `yarn learn run build` doesn't match the `build:integration` key.